### PR TITLE
model: receivers' permissions

### PR DESCRIPTION
**Pull request by hachreak**
(closes #51) (closes #53)

Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,13 +22,15 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
+dist: trusty
+
 addons:
-  postgresql: 9.4
+  postgresql: 9.6
```
Comment (lnielsen) [inline .travis.yml]: Why this change? We have many production system still on 9.4

File: setup.py [hunk 2]
```diff
@@ -76,7 +76,10 @@
 
 install_requires = [
     'Flask-BabelEx>=0.9.2',
+    'Flask-Security-Fork>=1.8.0',
```
Comment (lnielsen) [inline setup.py]: Can these dependencies not be handle by Invenio-DB and Invenio-Accounts?

File: .travis.yml (hunk 1 above)
Comment (hachreak) [inline .travis.yml]: I did long time ago, but I think was related to `JSONB` and the queries I can do with it.
I have to check again to be sure if it's really needed.

Comment (hachreak) [inline .travis.yml]: I tried now with postgresql 9.4 and I have an issue in alembic:

```
=================================== FAILURES ===================================
_________________________________ test_alembic _________________________________

app = <Flask 'testapp'>

    def test_alembic(app):
        """Test alembic recipes."""
        ext = app.extensions['invenio-db']
    
        with app.app_context():
            if db.engine.name == 'sqlite':
                raise pytest.skip('Upgrades are not supported on SQLite.')
    
            assert not ext.alembic.compare_metadata()
            db.drop_all()
>           ext.alembic.upgrade()

tests/test_invenio_webhooks.py:66: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/flask_alembic/extension.py:327: in upgrade
    self.run_migrations(do_upgrade)
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/flask_alembic/extension.py:236: in run_migrations
    env.run_migrations(**kwargs)
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/alembic/runtime/environment.py:817: in run_migrations
    self.get_context().run_migrations(**kw)
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/alembic/runtime/migration.py:329: in run_migrations
    step.migration_fn(**kw)
invenio_webhooks/alembic/ab9601b524f2_replace_json_columns_with_jsonb.py:60: in upgrade
    existing_type=existing_json_column())
<string>:8: in alter_column
    ???
<string>:3: in alter_column
    ???
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/alembic/operations/ops.py:1430: in alter_column
    return operations.invoke(alt)
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/alembic/operations/base.py:318: in invoke
    return fn(self, operation)
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/alembic/operations/toimpl.py:53: in alter_column
    **operation.kw
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/alembic/ddl/postgresql.py:104: in alter_column
    existing_nullable=existing_nullable,
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/alembic/ddl/impl.py:118: in _exec
    return conn.execute(construct, *multiparams, **params)
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/sqlalchemy/engine/base.py:945: in execute
    return meth(self, multiparams, params)
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/sqlalchemy/sql/ddl.py:68: in _execute_on_connection
    return connection._execute_ddl(self, multiparams, params)
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/sqlalchemy/engine/base.py:1002: in _execute_ddl
    compiled
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/sqlalchemy/engine/base.py:1189: in _execute_context
    context)
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/sqlalchemy/engine/base.py:1393: in _handle_dbapi_exception
    exc_info
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/sqlalchemy/util/compat.py:203: in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/sqlalchemy/util/compat.py:186: in reraise
    raise value.with_traceback(tb)
../../../virtualenv/python3.5.2/lib/python3.5/site-packages/sqlalchemy/engine/base.py:1182: in _execute_context
    context)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7fa66a5c8c18>
cursor = <cursor object at 0x7fa670e28140; closed: -1>
statement = 'ALTER TABLE webhooks_events ALTER COLUMN payload TYPE JSONB '
parameters = {}
context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x7fa66a6a2a20>

    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) column "payload" cannot be cast automatically to type jsonb
E       HINT:  You might need to specify "USING payload::jsonb".
E        [SQL: 'ALTER TABLE webhooks_events ALTER COLUMN payload TYPE JSONB ']
```

Comment (lnielsen) [inline .travis.yml]: But then we will have this error when we try to upgrade the production instances. Do you have an idea how to fix it or where it is coming from?

Comment (hachreak) [inline .travis.yml]: I found that postgresql v9.4 requires a specific configuration `postgresql_using='payload::jsonb',` that is not required on v9.6.
But it looks working. So, I'll downgrade to the version 9.4.
Now the only issue is mysql, with:
`AttributeError: 'function' object has no attribute '_compiler_dispatch'`

**Comment by hachreak** [comment]
ping @egabancho

**Comment by lnielsen** [comment]
Closing PR because it's been stale for a long time.
